File: DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/Inbound/BinaryParsing.cs
using ConformingContainerAntipattern._3_ConformingContainer.InMessages;
using ConformingContainerAntipattern._3_ConformingContainer.Interfaces;

namespace ConformingContainerAntipattern._3_ConformingContainer.Inbound
{
  internal interface IPacketParsing
  {
    AcmeMessage ResultFor(byte[] frameData);
  }

  class BinaryParsing : IPacketParsing
  {
    public AcmeMessage ResultFor(byte[] frameData)
    {
      if (frameData == null)
      {
        return ApplicationRoot.Context.Resolve<NullMessage>();
      }
      else if (frameData[0] == 1)
      {
        return ApplicationRoot.Context.Resolve<StartMessage>();
      }
      else
      {
        return ApplicationRoot.Context.Resolve<StopMessage>();
      }
      
    }
  }
}

File: Java/dependency-injection/_06_conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/Inbound/BinaryParsing.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.Inbound;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.ApplicationRoot;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.NullMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.StartMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages.StopMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Interfaces.InboundMessage;

public class BinaryParsing implements PacketParsing {
  public InboundMessage resultFor(byte[] frameData) {
    if (frameData == null) {
      return ApplicationRoot.CONTEXT.resolve(NullMessage.class);
    } else if (frameData[0] == 1) {
      return ApplicationRoot.CONTEXT.resolve(StartMessage.class);
    } else {
      return ApplicationRoot.CONTEXT.resolve(StopMessage.class);
    }

  }
}
